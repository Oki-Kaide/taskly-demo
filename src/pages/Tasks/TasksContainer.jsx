import React from 'react';
import PropTypes from 'prop-types';
import Tasks from './Tasks';

class TasksContainer extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  render() {
    const { accountData, logout } = this.props;
    return(
      <Tasks accountData={accountData} logout={logout} />
    );
  }
}

TasksContainer.propTypes = {
  logout: PropTypes.func.isRequired,
  accountData: PropTypes.shape().isRequired,
}
export default TasksContainer;