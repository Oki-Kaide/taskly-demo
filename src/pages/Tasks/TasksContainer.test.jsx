import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TasksContainer from './TasksContainer';

const createTestProps = props => ({
  accountData: {
    name: 'test',
    avatar: '',
  },
  logout: jest.fn(),
  ...props,
});

describe('TasksContainer component', () => {
  let props;

  beforeEach(() => {
    props = createTestProps({});
  });

  it('renders without crashing', () => {
    shallow(<TasksContainer {...props} />);
  });

  it('matches the rendered snapshot', () => {
    const component = shallow(<TasksContainer {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
