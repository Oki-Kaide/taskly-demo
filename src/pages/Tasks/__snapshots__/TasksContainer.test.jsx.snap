// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksContainer component matches the rendered snapshot 1`] = `
<Tasks
  accountData={
    Object {
      "avatar": "",
      "name": "test",
    }
  }
  logout={[MockFunction]}
/>
`;
